### PR TITLE
Feat/add product 1-4 connect

### PR DESCRIPTION
**Pull request by Blaxwo**
## Describe your changes

- I created new 4 step for adding prosuct
- I connected with backend 1, 4 steps
- I changed logic for going from different steps in add product form

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-229?atlOrigin=eyJpIjoiMTEzZjAxOTZmMTE2NGJiYmEyZDA0MzU2Y2VmM2QzYzEiLCJwIjoiaiJ9)
-  [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-220?atlOrigin=eyJpIjoiY2I4NzdmNzc4ZmM2NDliYmJjZDQxM2NhODM3NWEwYzciLCJwIjoiaiJ9)
- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-227?atlOrigin=eyJpIjoiYTQ1MGVmMGZiNWEwNGUyOTg2N2Q5ZDc3MmQ0YzE3YmEiLCJwIjoiaiJ9)

### **General**

- [ ] Assigned myself to the PR
- [ ] Assigned the appropriate labels to the PR
- [ ] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [ ] Performed a self-review of my code
- [ ] Types for input and output parameters
- [ ] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [ ] Don't have magic numbers
- [ ] Compare only with constants not with strings
- [ ] No ternary operator inside the ternary operator
- [ ] Don't have commented code
- [ ] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [ ] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [ ] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [ ] Components and business logic are separated
- [ ] Colors, Font Size, and Font Name is on the theme or in the constants
- [ ] No text in the components, use i18n approach
- [ ] No inline styles
- [ ] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.
зображення

### CODE REVIEW

File: src/components/shared/ImageCard/index.tsx [hunk 1]
```diff
@@ -9,42 +10,161 @@ import Primary from 'src/assets/icons/addProduct/primary.svg';
...
+    if (file) {
+      const isValid = await validateImage(file);
+
+      if (isValid) {
+        await sendPhotoRequest(file);
+      }
+    }
+  };
+
   return (
```
Comment (lyudad) [inline src/components/shared/ImageCard/index.tsx:35]: all above move to custom hook

File: src/components/shared/ImageCard/index.tsx [hunk 2]
```diff
@@ -9,42 +10,161 @@ import Primary from 'src/assets/icons/addProduct/primary.svg';
...
+
+    if (file) {
+      const isValid = await validateImage(file);
+
+      if (isValid) {
+        await sendPhotoRequest(file);
+      } else {
+        setPendingSrc(null);
+      }
     }
```
Comment (lyudad) [inline src/components/shared/ImageCard/index.tsx]: try/catch

File: src/components/shared/ImageCard/index.tsx (hunk 1 above)
Comment (Blaxwo) [reply src/components/shared/ImageCard/index.tsx:35]: done

File: src/components/shared/ImageCard/index.tsx (hunk 2 above)
Comment (Blaxwo) [reply src/components/shared/ImageCard/index.tsx]: done

**Pull request merged**